### PR TITLE
Release V 0.14.2

### PR DESCRIPTION
**Pull request by cirrusasf**
This release does followings:
* `--publish-bucket` option has been added to the HyP3 entry point to publish product to the ITS_LIVE AWS Open Data bucket, `s3://its-live-data` or the test bucket `s3://its-live-data-test`.
* `upload_file_to_s3_with_upload_access_keys` to perform S3 uploads with credentialed S3 clients.
* use of `UPLOAD_ACCESS_KEY_ID` and `UPLOAD_ACCESS_KEY_SECRET` to upload products to write-protected bucket.

### CODE REVIEW

**Comment by forrestfwilliams** [comment]
To run the container, first pull the dev version:
```
docker pull ghcr.io/asfhyp3/hyp3-autorift:0.14.2.dev32_g0323fa6
```
Then run it **specifying the test open data bucket!!!**
```
docker run -it --rm \
    -e EARTHDATA_USERNAME=[YOUR_USERNAME_HERE] \
    -e EARTHDATA_PASSWORD=[YOUR_PASSWORD_HERE] \
    -e ESA_USERNAME=[YOUR_USERNAME_HERE] \
    -e ESA_PASSWORD=[YOUR_PASSWORD_HERE] \
    -e UPLOAD_ACCESS_KEY_ID=[YOUR_ACCESS_KEY_ID] \
    -e UPLOAD_ACCESS_KEY_SECRET=[YOUR_ACCESS_KEY_SECRET] \
    ghcr.io/asfhyp3/hyp3-autorift:0.14.2.dev32_g0323fa6 \
    LC09_L1GT_215109_20220125_20230430_02_T2 LC09_L1GT_215109_20220210_20230428_02_T2 \
    --publish-bucket its-live-data-test
```

**Comment by forrestfwilliams** [comment]
You'll then need to check the test bucket to ensure that the data has been correctly uploaded.

**Comment by forrestfwilliams** [comment]
Have confirmed that running:
```bash
hyp3_autorift LC09_L1GT_215109_20220125_20230430_02_T2 LC09_L1GT_215109_20220210_20230428_02_T2 --publish-bucket its-live-data-test
```
With the correct environment variables set runs and uploads data to the test bucket.

There is an issue with the docker command above, but I'm guessing it uses the wrong version of the docker container.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/)
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.14.2]
+### Added
+* `--publish-bucket` option has been added to the HyP3 entry point to publish product to the ITS_LIVE AWS Open Data bucket, `s3://its-live-data` or the test bucket `s3://its-live-data-test`.
+* `upload_file_to_s3_with_upload_access_keys` to perform S3 uploads with credentialed S3 clients.
```
Comment (jtherrmann) [inline CHANGELOG.md:12]: Do users need to know about this function? If not, I suggest deleting this line from the Changelog.
```suggestion
```

Comment (forrestfwilliams) [reply CHANGELOG.md:12]: I think it's relevant. Other members of the ITS_LIVE team will be interested in it.

Comment (jtherrmann) [reply CHANGELOG.md:12]: Sounds good.

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**